src/main/java/appeng/integration/modules/helpers/JabbaBarrel.java

```diff
@@ -59,9 +59,23 @@ public IAEItemStack injectItems(final IAEItemStack input, final Actionable mode,
             {
                 return input;
             }
-            if( mode == Actionable.MODULATE )
+            long max = this.barrel.getMaxStoredCount();
```

Comment:
Barrel's getMaxStoredCount() is only correct when there are stuff inside it. Just like drawers, barrel's maximum capacity depends on the stack size of input item. If there is nothing inside, it'd just assume to use the capacity of last item placed inside.

Reply:
```	public int getMaxStoredCount() {
		return this.deleteExcess ? this.totalCapacity + this.stackAmount: this.totalCapacity;
	}
```

Reply:
totalCapcity will be changed to maxStackSize*maxStack when you put in the first item.

Reply:
Indeed

Reply:
fixed